mmpose/datasets/datasets/top_down/topdown_h36m_dataset.py

```diff
@@ -0,0 +1,297 @@
+        # padding to include proper amount of context
+        scale = scale * padding
+
+        return center, scale
+
+    def _get_db(self):
+        """Load dataset."""
+        gt_db = []
+        bbox_id = 0
+        num_joints = self.ann_info['num_joints']
+        for img_id in self.img_ids:
+
+            ann_ids = self.coco.getAnnIds(imgIds=img_id, iscrowd=False)
+            objs = self.coco.loadAnns(ann_ids)
+
+            for obj in objs:
+                if max(obj['keypoints']) == 0:
+                    continue
+                joints_3d = np.zeros((num_joints, 3), dtype=np.float32)
+                joints_3d_visible = np.zeros((num_joints, 3), dtype=np.float32)
+
+                keypoints = np.array(obj['keypoints']).reshape(-1, 3)
+                joints_3d[:, :2] = keypoints[:, :2]
+                joints_3d_visible[:, :2] = np.minimum(1, keypoints[:, 2:3])
+
+                # use 1.25 padded bbox as input
+                center, scale = self._xywh2cs(*obj['bbox'][:4])
+
+                image_file = os.path.join(self.img_prefix,
+                                          self.id2name[img_id])
+
+                gt_db.append({
+                    'image_file': image_file,
+                    'center': center,
+                    'scale': scale,
+                    'rotation': 0,
+                    'joints_3d': joints_3d,
+                    'joints_3d_visible': joints_3d_visible,
+                    'dataset': self.dataset_name,
+                    'bbox': obj['bbox'],
+                    'bbox_score': 1,
+                    'bbox_id': bbox_id
+                })
+                bbox_id = bbox_id + 1
+        gt_db = sorted(gt_db, key=lambda x: x['bbox_id'])
+
+        return gt_db
+
+    def evaluate(self, outputs, res_folder, metric, **kwargs):
+        """Evaluate human3.6m 2d keypoint results. The pose prediction results
+        will be saved in `${res_folder}/result_keypoints.json`.
+
+        Note:
+            batch_size: N
+            num_keypoints: K
+            heatmap height: H
+            heatmap width: W
+
+        Args:
+            outputs (list(dict))
+                :preds (np.ndarray[N,K,3]): The first two dimensions are
+                    coordinates, score is the third dimension of the array.
+                :boxes (np.ndarray[N,6]): [center[0], center[1], scale[0]
+                    , scale[1],area, score]
+                :image_paths (list[str]): For example, ['data/coco/val2017
+                    /000000393226.jpg']
+                :heatmap (np.ndarray[N, K, H, W]): model output heatmap
+                :bbox_id (list(int)).
+            res_folder (str): Path of directory to save the results.
+            metric (str | list[str]): Metric to be performed. Defaults: 'mAP'.
+
+        Returns:
+            dict: Evaluation results for evaluation metric.
+        """
+        metrics = metric if isinstance(metric, list) else [metric]
+        allowed_metrics = ['EPE']
```

Comment:
Why not support more metrics?

Reply:
Human3.6M is not a commonly used dataset for 2d keypoint detection, and we add this config and model mainly for two-stage 3d keypoint demos. So maybe extensive evaluation is unnecessary here?

Reply:
I suggest using PCK or AP as the default metrics.

Reply:
Updated to use PCK@0.05 as the default metrics